hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryDiskUsageOrdering.java

```diff
@@ -0,0 +1,421 @@
+        Files.createDirectory(temporaryFolder.resolve("JunitOmDBDir"))
+            .toFile());
+    OzoneManagerServiceProviderImpl ozoneManagerServiceProvider =
+        getMockOzoneManagerServiceProviderWithFSO();
+    reconOMMetadataManager = getTestReconOmMetadataManager(omMetadataManager,
+        Files.createDirectory(temporaryFolder.resolve("OmMetataDir")).toFile());
+
+    ReconTestInjector reconTestInjector =
+        new ReconTestInjector.Builder(temporaryFolder.toFile())
+            .withReconOm(reconOMMetadataManager)
+            .withOmServiceProvider(ozoneManagerServiceProvider)
+            .withReconSqlDb()
+            .withContainerDB()
+            .addBinding(OzoneStorageContainerManager.class,
+                getMockReconSCM())
+            .addBinding(StorageContainerServiceProvider.class,
+                mock(StorageContainerServiceProviderImpl.class))
+            .addBinding(NSSummaryEndpoint.class)
+            .build();
+    ReconNamespaceSummaryManager reconNamespaceSummaryManager =
+        reconTestInjector.getInstance(ReconNamespaceSummaryManager.class);
+    nsSummaryEndpoint = reconTestInjector.getInstance(NSSummaryEndpoint.class);
+
+    // populate OM DB and reprocess into Recon RocksDB
+    populateOMDB();
+    NSSummaryTaskWithFSO nSSummaryTaskWithFso =
+        new NSSummaryTaskWithFSO(reconNamespaceSummaryManager,
+            reconOMMetadataManager, ozoneConfiguration);
+    nSSummaryTaskWithFso.reprocessWithFSO(reconOMMetadataManager);
+  }
+
+  /**
+   * Create a new OM Metadata manager instance with one user, one vol, and two
+   * buckets.
+   * @throws IOException ioEx
+   */
+  private static OMMetadataManager initializeNewOmMetadataManager(
+      File omDbDir)
+      throws IOException {
+    OzoneConfiguration omConfiguration = new OzoneConfiguration();
+    omConfiguration.set(OZONE_OM_DB_DIRS,
+        omDbDir.getAbsolutePath());
+    OMMetadataManager omMetadataManager = new OmMetadataManagerImpl(
+        omConfiguration, null);
+    return omMetadataManager;
+  }
+
+  @Test
+  public void testDiskUsageOrderingForRoot() throws Exception {
+    // root level DU
+    // Verify the ordering of subpaths under the root
+    verifyOrdering(ROOT_PATH);
+  }
+
+  @Test
+  public void testDiskUsageOrderingForVolume() throws Exception {
+    // volume level DU
+    // Verify the ordering of subpaths under the volume
+    verifyOrdering("/volA");
+    verifyOrdering("/volB");
+  }
+
+  @Test
+  public void testDiskUsageOrderingForBucket() throws Exception {
+    // bucket level DU
+    // Verify the ordering of subpaths under the bucket
+    verifyOrdering("/volA/bucketA1");
+    verifyOrdering("/volA/bucketA2");
+    verifyOrdering("/volA/bucketA3");
+    verifyOrdering("/volB/bucketB1");
+  }
+
+  private void verifyOrdering(String path)
+      throws IOException {
+    Response response =
+        nsSummaryEndpoint.getDiskUsage(path, true, false);
```

Comment:
Is this API will be returning all data ? or this PR is only to just sort data in du response object ?
Because this PR title says that to display Top N records.

Reply:
I have added a limit parameter to return only the top N disk usage records.
Please check the value is placed inside the constant :- `DISK_USAGE_TOP_RECORDS_LIMIT`